gems/smithy/spec/interfaces/client/shapes_spec.rb

```diff
@@ -20,45 +20,47 @@
     expect(subject).to be_a(Module)
   end
 
-  context 'generated shapes' do
+  context 'shapes' do
     shape_tests['shapes'].each do |id, shape|
-      context 'a generated shape' do
+      context "a generated shape: #{id}" do
         next if %w[operation service].include?(shape['type'])
 
         let(:shape_name) { Smithy::Vise::Shape.relative_id(id) }
         let(:generated_shape) { Object.const_get("#{subject}::#{shape_name}") }
 
-        it 'is a shape of expected shape type and has an id' do
+        it 'is of a expected shape type and has an id' do
           expected_shape_type =
             Object.const_get("#{shapes_module}::#{shape['type'].camelize}Shape")
 
           expect(generated_shape).to be_a(expected_shape_type)
           expect(generated_shape.id).to eq(id)
         end
 
-        it 'has a type variation of the shape when applicable' do
-          unless %w[structure union].include?(shape['type'])
-            skip("Test does not expect the generated #{id} to have a type")
-          end
+        context 'type' do
+          next unless %w[structure union].include?(shape['type'])
```

Comment:
general style thought - rather than always defining this context, maybe use an if outside of it?  IE:

```ruby
if %w[structure union].include?(shape['type'])
  context 'type' do
    it 'is set on the shape' .... 
  end
end
```

Reply:
Yes you would have to do that above too. Otherwise you get an empty context section.